### PR TITLE
feat(jdk19): add a jdk19 container

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3243, provide a docker image for jdk19

another PR to follow with the updatecli manifest.

### CODE REVIEW

**Comment by dduportal** [comment]
`powershell.exe : manifest for jenkins/inbound-agent:jdk19-nanoserver-1809 not found: manifest unknown: manifest unknown` you might want to use the "jdk11" image :)

**Comment by smerle33** [comment]
> `powershell.exe : manifest for jenkins/inbound-agent:jdk19-nanoserver-1809 not found: manifest unknown: manifest unknown` you might want to use the "jdk11" image :)

the core image stay the jdk11 one but the other one should vary if I take example on jdk8 and jdk17 ...

**Review by dduportal — APPROVED** [review]

File: maven/jdk19/Dockerfile.nanoserver [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# escape=`
+# this JAVA version is the one used for the execution of the agent process and must match the controller version
+ARG JAVA_VERSION=19.0.1_10
+FROM eclipse-temurin:${JAVA_VERSION}-jdk-windowsservercore-1809 AS core
+FROM jenkins/inbound-agent:jdk11-nanoserver-1809
+
+# ProgressPreference => Disable Progress bar for faster downloads
+SHELL ["pwsh.exe", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# Retrieve JDK11 for running jenkins agent process but do not use it as default
```
Comment (lemeurherve) [inline maven/jdk19/Dockerfile.nanoserver:10]: JDK19?

File: maven/jdk19/cst-windows.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+schemaVersion: 2.0.0
+
+commandTests:
+  - name: "Check that `java` is present in the PATH and match jdk19"
+    command: "cmd.exe"
+    args: ["/C", "java", "--version"]
+    expectedOutput: ["Temurin-19"]
+  - name: "Check that `java 11` is present in C:/openjdk-11"
+    command: "cmd.exe"
+    args: ["/C", "C:/openjdk-11/bin/java", "--version"]
```
Comment (lemeurherve) [inline maven/jdk19/cst-windows.yml:10]: Same, should test jdk19 and not jdk11?

Comment (smerle33) [reply maven/jdk19/cst-windows.yml:10]: no, jdk11 used to spawn the agent and match the jdk used on the controller.
and the jdk19 installed in the /tools folder and the path, for default used from the pipeline (with java)

**Pull request merged**